mentat/sampler/README.md

```diff
@@ -0,0 +1,23 @@
+# Mentat Sampler API
+The Sampler API is an open-source standard to captures interactions between a developer and an LLM-based AI Assistant. It's intended to facilitate sharing of benchmarks and fine-tuning data throughout the open-source community and industry.
```

Comment:
I was a bit confused by this and I work on mentat; I think it would be good if the description was a bit clearer. Maybe we could give a description of what it actually is? (A sample contains the git diff of an edit made by an LLM with the user and assistant messages leading up to that edit)

Reply:
Also, I think it would be a lot simpler to just point to a git commit on github with the specified diff instead of having diff_edit, merge_base, diff_merge_base, and other arguments. Is there a reason that wouldn't work as well?

Reply:
I'm mostly concerned with being able to generate samples as I work, rather than having to sit down and 'make samples'. In practice I'm hardly ever working directly from a permanent commit because we squash-merge everything. They certainly can be blank, and maybe they usually will be.

Reply:
I'm pretty sure the commits will still be there (even if they are squash-merged). Git squashes all the commits into one commit and puts it on main, but the commits are still valid and can still be accessed via their hash.

Reply:
> Still though if you're working from a branch or commit that _could_ get abandoned, and you want to create a sample, it'd be nice to have this option to make sure it'll always be valid.

Maybe, but as far as I know, git will only ever delete a commit if the user does it (via a command), which I don't think we should worry about. If the commit is taken down off of it's origin (like github), then odds are all the other commits also were, in which case this entire sample is moot anyways.

Reply:
By abandoned I mean never pushed to GH/remote in the first place. Then it'd only available on my machine, right?

Reply:
Oh, I see, yeah.